Realm/RLMSyncConfiguration.mm

```diff
@@ -270,32 +286,44 @@ - (BOOL)enableFlexibleSync {
     return make_sync_error(errorClass, @(error.message.c_str()), error.error_code.value(), custom);
 }
 
-static void setDefaults(SyncConfig& config, RLMUser *user) {
-    config.client_resync_mode = ClientResyncMode::Manual;
-    config.stop_policy = SyncSessionStopPolicy::AfterChangesUploaded;
-
+- (void)assignConfigErrorHandler:(RLMUser *)user {
     RLMSyncManager *manager = [user.app syncManager];
     __weak RLMSyncManager *weakManager = manager;
-    config.error_handler = [weakManager](std::shared_ptr<SyncSession> errored_session, SyncError error) {
+    RLMSyncErrorReportingBlock resetHandler = self.manualClientResetHandler;
+    _config->error_handler = [weakManager, resetHandler](std::shared_ptr<SyncSession> errored_session, SyncError error) {
         RLMSyncErrorReportingBlock errorHandler;
         @autoreleasepool {
             errorHandler = weakManager.errorHandler;
         }
```

Comment:
I think consolidating the decision on which callback to use at the start makes this function a little simpler. It also avoids capturing the unused callback, which probably doesn't really matter.

```suggestion
    _config->error_handler = [weakManager, resetHandler](std::shared_ptr<SyncSession> errored_session, SyncError error) {
        RLMSyncErrorReportingBlock errorHandler;
        if (error.is_client_reset_requested()) {
            errorHandler = resetHandler;
        }
        if (!errorHandler) {
            @autoreleasepool {
                errorHandler = weakManager.errorHandler;
            }
        }
        if (!errorHandler) {
            return;
        }
```